### PR TITLE
(GH-537) Adding support for: bolt group show

### LINKED ISSUE #537: list nodes and groups
To allow convenient inspection of group/node definitions  bolt should support a "list" and "show" option for groups and nodes.

```
$ bolt group list
mygroup-01
mygroup-02
mygroup-03
$ bolt group show mygroup-02
foo-001,foo-002,foo-003
$ bolt group show mygroup-02 --with-spaces
foo-001 foo-002 foo-003
$ bolt group show mygroup-02 --with-newlines
foo-001
foo-002
foo-003

$ bolt group show all --with-newlines
foo-001
foo-002
foo-003
...
foo-400
```

This is a precondition to add host and group completion for shell completions like "./resources/bolt_bash_completion.sh"
See also: https://github.com/puppetlabs/bolt/issues/495

### PR DESCRIPTION
**Pull request by nmaludy**
Work on `bolt group show` support to list available groups in the inventory file.

May also be adding support for additional sub commands.

**TODO**
- [x] Unit tests
- [x] Integration tests

Closes #537

### CODE REVIEW

File: lib/bolt/bolt_option_parser.rb [hunk 1]
```diff
@@ -7,6 +7,7 @@
 module Bolt
   class BoltOptionParser < OptionParser
     OPTIONS = { inventory: %w[nodes targets query rerun description],
+                group: %w[nodes targets query rerun description],
```
Comment (lucywyman) [inline lib/bolt/bolt_option_parser.rb]: I don't think this command will actually do anything with these options. I'd remove this, and modify the line below to not include it.

File: lib/bolt/bolt_option_parser.rb [hunk 2]
```diff
@@ -31,6 +32,9 @@ def get_help_text(subcommand, action = nil)
       when 'inventory'
         { flags: OPTIONS[:inventory] + OPTIONS[:global] + %w[format inventoryfile boltdir configfile],
           banner: INVENTORY_HELP }
+      when 'group'
+        { flags: OPTIONS[:group] + OPTIONS[:global] + %w[format inventoryfile boltdir configfile],
```
Comment (lucywyman) [inline lib/bolt/bolt_option_parser.rb]:
```suggestion
        { flags:  OPTIONS[:global] + %w[format inventoryfile boltdir configfile],
```

File: spec/integration/inventory_spec.rb [hunk 3]
```diff
@@ -392,4 +393,42 @@ def fact_plan(name = 'facts_test')
...
+                  config: {
+                    transport: conn[:protocol],
+                    conn[:protocol] => {
+                      user: conn[:user],
+                      port: conn[:port]
+                    }
+                  } }
+              ] }
+          ],
+        config: {
```
Comment (lucywyman) [inline spec/integration/inventory_spec.rb]: I think you can remove from here to line 426

**Comment by lucywyman** [comment]
This should also include:
- [ ] short changelog update
- [ ] add the `group show` command to the command reference page (which we really should be generating from help text but alas are not :cry:) https://github.com/puppetlabs/bolt/blob/master/documentation/bolt_command_reference.md

**Comment by nmaludy** [comment]
@lucywyman should be good to go!

**Pull request merged**